src/registry/file/file-tab.tsx

```diff
@@ -0,0 +1,263 @@
+      valueProperty="value"
+      onChange={onChange}
+    />
+  );
+};
+
+const UseConfigFiletypes = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'useConfigFiletypes' builder field",
+    defaultMessage:
+      'When this is checked, the filetypes configured in the global settings will be used.',
+  });
+  return (
+    <Checkbox
+      name="useConfigFiletypes"
+      label={
+        <FormattedMessage
+          description="Label for 'useConfigFiletypes' builder field"
+          defaultMessage="Use globally configured filetypes"
+        />
+      }
+      tooltip={tooltip}
+    />
+  );
+};
+
+const FileMaxSize = () => {
+  const intl = useIntl();
+  const {serverUploadLimit} = useContext(BuilderContext);
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'fileMaxSize' builder field",
+    defaultMessage: 'The maximum size of a single file to upload.',
+  });
+
+  return (
+    <TextField
+      name="fileMaxSize"
+      label={
+        <FormattedMessage
+          description="Label for 'fileMaxSize' builder field"
+          defaultMessage="Maximum file size"
+        />
+      }
+      tooltip={tooltip}
+      description={
+        <FormattedMessage
+          description="Description for 'fileMaxSize' builder field"
+          defaultMessage="Note that the server upload limit is {serverUploadLimit}."
+          values={{serverUploadLimit}}
+        />
+      }
+    />
+  );
+};
+
+const MaxNumberOfFiles = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'maxNumberOfFiles' builder field",
+    defaultMessage: 'The maximum number of files that can be uploaded.',
+  });
+  return (
+    <NumberField
+      name="maxNumberOfFiles"
+      label={
+        <FormattedMessage
+          description="Label for 'maxNumberOfFiles' builder field"
+          defaultMessage="Maximum number of files"
+        />
+      }
+      tooltip={tooltip}
+      min={1}
+      step={1}
+    />
+  );
+};
+
+const ImageResizeApply = () => {
+  const intl = useIntl();
+  const tooltip = intl.formatMessage({
+    description: "Tooltip for 'of.image.resize.apply' builder field",
+    defaultMessage: 'When this is checked, any uploaded image(s) will be resized.',
+  });
+  return (
+    <Checkbox
+      name="of.image.resize.apply"
+      label={
+        <FormattedMessage
+          description="Label for 'of.image.resize.apply' builder field"
+          defaultMessage="Resize image"
+        />
+      }
+      tooltip={tooltip}
+    />
+  );
+};
+
+const ImageResizingOptions = () => {
+  const {values} = useFormikContext<FileComponentSchema>();
+  const applyResize = values?.of?.image?.resize?.apply ?? false;
+
+  return (
+    <>
+      <ImageResizeApply />
+      {applyResize && (
+        <Component type="columns">
+          <div className="columns">
+            <div className="column column--span-md">
```

Comment:
column styles are in the backend CSS, so they're displayed in a single column in storybook